src/components/controller/controller.ts

```diff
@@ -0,0 +1,40 @@
+import AppLoader from './appLoader';
+import { DataNews, DataSources, Callback } from '../options';
+
+class AppController extends AppLoader {
+  getSources(callback: Callback<DataSources>) {
+    super.getResp(
+      {
+        endpoint: 'sources',
+      },
+      callback
+    );
+  }
+
+  getNews(e: Event, callback: Callback<DataNews>) {
+    let target = e.target;
```

Comment:
Опять же ниже не придется подписывать as HTMLelement несколько раз
```suggestion
    let target = e.target as HTMLElement;
```